### ISSUE TITLE
Optimize comparison

### ISSUE THREAD

**Issue by fabiolimace** [f4b6a3/ksuid-creator#14]

**Comment by fabiolimace** [comment]
Benchmark results before the changes:

```
Benchmark              Mode  Cnt      Score      Error   Units
Throughput.compareTo  thrpt    5  68817,326 ± 1647,961  ops/ms
```

Benchmark results after the changes:

```
Benchmark              Mode  Cnt      Score     Error   Units
Throughput.compareTo  thrpt    5  76785,167 ± 252,614  ops/ms (1.115785)
```

Benchmark code:

```java

package benchmark;

import java.util.concurrent.TimeUnit;

import com.github.f4b6a3.ksuid.Ksuid;
import com.github.f4b6a3.ksuid.KsuidCreator;

import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.BenchmarkMode;
import org.openjdk.jmh.annotations.Fork;
import org.openjdk.jmh.annotations.Measurement;
import org.openjdk.jmh.annotations.Mode;
import org.openjdk.jmh.annotations.OutputTimeUnit;
import org.openjdk.jmh.annotations.Scope;
import org.openjdk.jmh.annotations.State;
import org.openjdk.jmh.annotations.Threads;
import org.openjdk.jmh.annotations.Warmup;

@Fork(1)
@Threads(1)
@State(Scope.Benchmark)
@Warmup(iterations = 5, time = 1)
@Measurement(iterations = 5, time = 3)
@BenchmarkMode(Mode.Throughput)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
public class Throughput {

	Ksuid ksuid1 = new Ksuid(0, new byte[16]);
	Kadui ksuid2 = new Ksuid(0, new byte[16]);
	
	@Benchmark
	public int compareTo() {
		return ksuid1.compareTo(ksuid2);
	}
}
```

**Comment by fabiolimace** [comment]
Released v4.0.1
